apps/pg_subscriptor/lib/column.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Column do
+
+  @type kind :: ?n | ?u | ?t | ?b
+  @type t :: %__MODULE__{
+    kind: kind,
+    value: nil | binary,
+  }
+
+  defstruct [:kind, :value]
+
+  def get_col(0, data) do
+    {[], data}
+  end
+
+  def get_col(num_of_cols, data) do
+    <<col_type::utf8, rest::binary>> = data
+
+    {value, rest} = if col_type == ?b do
+      <<col_length::32, rest::binary>> = rest
+      <<value::binary-size(col_length), rest::binary>> = rest
+      {value, rest}
+    else
+      {nil, rest}
+    end
```

Comment:
Since we will have to support all kinds of column types (encodings) I suggest changing if to case.

Note, when update is called on the example database the column type is `t`.

Reply:
I understood it that it will be present for all and it will be either `n`, `b`, `t`, or `u`. And the length of column value N and binary of length N.

Reply:
So, for example, for `n`, the length will be `0` and the value will be omitted?

I understood it as it will be `n`/`t`/`u` alone or `b` with `length` and `value`.
That is why I said that the doc is confusing...

Reply:
Yep, but it kind of makes sense. `n` signals null-terminated String so we don't need to know the length. `u` I don't understand yet, not sure what are TOAST bytes.